orbit/cmd/orbit/orbit.go

```diff
@@ -1244,9 +1245,27 @@ func (d *desktopRunner) execute() error {
 		opts = append(opts, execuser.WithEnv("FLEET_DESKTOP_INSECURE", "1"))
 	}
 
+	var buf bytes.Buffer
+
 	for {
 		// First retry logic to start fleet-desktop.
 		if done := retry(30*time.Second, false, d.interruptCh, func() bool {
+			wcmd := exec.Command("/usr/bin/stat", "-f", "%Su", "/dev/console")
+			wcmd.Stderr = os.Stderr
+			wcmd.Stdout = &buf
+			if err := wcmd.Run(); err != nil {
+				log.Error().Err(err).Msg("calling stat")
+			}
+
+			// If no user is logged in via GUI, the command line returns "root".
+			if strings.TrimSpace(buf.String()) == "root" {
+				log.Debug().Msg("no user found logged into GUI")
+				buf.Reset()
+				return true
+			}
+
+			buf.Reset()
```

Comment:
I think this whole part that checks for the GUI user is macOS-specific, right? It shouldn't run on other platforms? If so this should be a function call and that function be defined as a no-op for other platforms, and do the `stat` call only for macOS. You can see examples of how to do that in the `orbit/pkg/execuser` sub-package (you'll see the same function defined in different files, some with the `_darwin` suffix, and in that file there's a Go directive comment at the top saying to build it only on such platform).

I think this function could go in the `orbit/pkg/platform` package? Seems to make sense to me, but if you think there's another better place go for it.

Reply:
I went ahead and created a new `user` package as the existing `platform` package was structured in a way that I'd have to make a bunch of edits and that would expand the testing scope for this PR.